cf-rd/src/main/java/org/eclipse/californium/tools/resources/RDNodeResource.java

```diff
@@ -179,7 +180,7 @@ private void setContextFromRequest(Request request, String newContext)
 	 * humidity. If the path is /readings/temp, temp will be a subResource
 	 * of readings, which is a subResource of the node.
 	 */
-	public CoapResource addNodeResource(String path) {
+	public CoapResource addResource(String path, ResourceAttributes attributes) {
```

Comment:
I do not understand why you want to pass in the attributes here? you can simply set the attributes on the returned `CoapResource`:
````
CoapResouce newRes = addNodeResouce("my/path");
newRes.getAttributes().addResourceType("myType");
...
````